tests/integration/test_ingress_relation.py

```diff
@@ -55,13 +64,16 @@ def start_server(self):
         deploy_any_charm(json.dumps(src_overwrite)),
         build_and_deploy_ingress(),
     )
+
     await ingress.set_config({"service-hostname": "any"})
     await model.wait_for_idle()
-    await run_action("any", "rpc", method="start_server")
     await model.add_relation("any:ingress", "ingress:ingress")
-    await model.wait_for_idle()
+    await model.wait_for_idle(status="active")
+    await run_action("any", "rpc", method="start_server")
+
     response = requests.get(
         f"http://127.0.0.1/{model.name}-any/ok", headers={"Host": "any"}, timeout=5
     )
-    assert response.text == "ok"
+
+    assert response.text.rpartition("/")[0] == "http://any"
```

Comment:
Have discussed this a bit outside this PR and it seems like we're doing this because we're seeing random path-routes values. We should try and figure out where those are coming from and why we can't set it to a specific value in this test.